pinot-segment-local/src/main/java/org/apache/pinot/segment/local/io/util/FixedByteValueReaderWriter.java

```diff
@@ -92,6 +92,13 @@ public String getPaddedString(int index, int numBytesPerValue, byte[] buffer) {
     return new String(buffer, 0, numBytesPerValue, UTF_8);
   }
 
+  public void getCustomPaddedString(int offset, int numBytesPerValue, ByteBuffer buffer) {
+    long l1 = _dataBuffer.getLong(offset);
```

Comment:
Diving into Java instruction set, getLong seems to always resolve into `invokevirtual`, which means there's no chance of auto vectorization here (except for offset computation). Kinda surprising to me how sub-optimal all of this is